mm2src/coins/lp_coins.rs

```diff
@@ -2707,23 +2739,103 @@ impl Default for PrivKeyActivationPolicy {
+pub struct EthMetamaskPolicy {
+    pub(crate) public_key: EthH264,
+    pub(crate) public_key_uncompressed: EthH520,
+}
+
+impl<T> From<T> for PrivKeyPolicy<T> {
+    fn from(key_pair: T) -> Self { PrivKeyPolicy::Iguana(key_pair) }
 }
 
 impl<T> PrivKeyPolicy<T> {
-    pub fn key_pair(&self) -> Option<&T> {
+    pub fn activated_key(&self) -> Option<&T> {
         match self {
-            PrivKeyPolicy::KeyPair(key_pair) => Some(key_pair),
+            PrivKeyPolicy::Iguana(key_pair) => Some(key_pair),
+            PrivKeyPolicy::HDWallet {
+                activated_key: activated_key_pair,
+                ..
+            } => Some(activated_key_pair),
             PrivKeyPolicy::Trezor => None,
+            #[cfg(target_arch = "wasm32")]
+            PrivKeyPolicy::Metamask(_) => None,
         }
     }
 
-    pub fn key_pair_or_err(&self) -> Result<&T, MmError<PrivKeyPolicyNotAllowed>> {
-        self.key_pair()
-            .or_mm_err(|| PrivKeyPolicyNotAllowed::HardwareWalletNotSupported)
+    pub fn activated_key_or_err(&self) -> Result<&T, MmError<PrivKeyPolicyNotAllowed>> {
+        self.activated_key().or_mm_err(|| {
+            PrivKeyPolicyNotAllowed::UnsupportedMethod(
+                "`key_pair_or_err` is supported only for `PrivKeyPolicy::KeyPair` or `PrivKeyPolicy::HDWallet`"
+                    .to_string(),
+            )
+        })
+    }
+
+    pub fn bip39_secp_priv_key(&self) -> Option<&ExtendedPrivateKey<secp256k1::SecretKey>> {
+        match self {
+            PrivKeyPolicy::HDWallet {
+                bip39_secp_priv_key, ..
+            } => Some(bip39_secp_priv_key),
+            PrivKeyPolicy::Iguana(_) | PrivKeyPolicy::Trezor => None,
+            #[cfg(target_arch = "wasm32")]
+            PrivKeyPolicy::Metamask(_) => None,
+        }
+    }
+
+    pub fn bip39_secp_priv_key_or_err(
+        &self,
+    ) -> Result<&ExtendedPrivateKey<secp256k1::SecretKey>, MmError<PrivKeyPolicyNotAllowed>> {
+        self.bip39_secp_priv_key().or_mm_err(|| {
+            PrivKeyPolicyNotAllowed::UnsupportedMethod(
+                "`bip39_secp_priv_key_or_err` is supported only for `PrivKeyPolicy::HDWallet`".to_string(),
+            )
+        })
+    }
+
+    pub fn derivation_path(&self) -> Option<&StandardHDPathToCoin> {
+        match self {
+            PrivKeyPolicy::HDWallet { derivation_path, .. } => Some(derivation_path),
+            PrivKeyPolicy::Iguana(_) | PrivKeyPolicy::Trezor => None,
+            #[cfg(target_arch = "wasm32")]
+            PrivKeyPolicy::Metamask(_) => None,
+        }
+    }
+
+    pub fn derivation_path_or_err(&self) -> Result<&StandardHDPathToCoin, MmError<PrivKeyPolicyNotAllowed>> {
+        self.derivation_path().or_mm_err(|| {
+            PrivKeyPolicyNotAllowed::UnsupportedMethod(
+                "`derivation_path_or_err` is supported only for `PrivKeyPolicy::HDWallet`".to_string(),
+            )
+        })
+    }
+
+    pub fn hd_wallet_derived_priv_key_or_err(
```

Comment:
> It's better to move the pub(crate) one up to the other pub method in the beginning of impl and leave private ones here

All methods in `impl<T> PrivKeyPolicy<T>` can be private. I will make them that way.

> hd_wallet_derived_priv_key also looks great because it returns a Result, there is no need to emphasize this fact in a function name

I prefer to keep the naming convention in this implementation consistent.

> hd_wallet already contains word "derived" in it (hierarchical derived), perhaps hd_private_key could be better

`hd_private_key` can refer to multiple things (e.g. bip39_secp_priv_key, etc..), that's why I used derived here.